### PR TITLE
Update baseline test yaml

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205
- [x] 202305

#### What is the motivation for this PR?
Jobs template yaml could not start with variables
#### How did you do it?
Set variables under stage

### CODE REVIEW

File: .azure-pipelines/baseline_test/baseline.test.mgmt.public.master.yml [hunk 1]
```diff
@@ -28,6 +28,13 @@ stages:
   dependsOn:
   - Pre_test
   condition: succeeded('Pre_test')
+  variables:
+  - group: SONiC-Elastictest
+  - name: inventory
+    value: veos_vtb
+  - name: testbed_file
+    value: vtestbed.yaml
+  - group: GIT_SECRETS
```
Comment (lerry-lee) [inline .azure-pipelines/baseline_test/baseline.test.mgmt.public.master.yml:37]: All Test_round_1, Test_round_2 and Test_round_3 have the same variables, can the definition of these variables be extracted to parent index?

File: .azure-pipelines/baseline_test/baseline.test.template.yml [hunk 2]
```diff
@@ -1,19 +1,11 @@
...
-
 jobs:
 - job: t0_elastictest
   displayName: "kvmtest-t0 by Elastictest"
   timeoutInMinutes: 240
   continueOnError: false
   pool: ubuntu-20.04
   strategy:
-    parallel: $(BASELINE_MGMT_PUBLIC_MASTER_TEST_NUM)
+    parallel: $[ variables['BASELINE_MGMT_PUBLIC_MASTER_TEST_NUM'] ]
```
Comment (lerry-lee) [inline .azure-pipelines/baseline_test/baseline.test.template.yml:8]: I wonder where is the definition of this variable `BASELINE_MGMT_PUBLIC_MASTER_TEST_NUM` ?

Comment (xwjiang-ms) [reply .azure-pipelines/baseline_test/baseline.test.template.yml:8]: Yes, and parallel variable type should be number, but our env imported would be string

File: .azure-pipelines/baseline_test/baseline.test.mgmt.public.master.yml (hunk 1 above)
Comment (xwjiang-ms) [reply .azure-pipelines/baseline_test/baseline.test.mgmt.public.master.yml:37]: No, even defined in previous step, we also need to import to different stages, maybe we can create a template file and import template, but I think it's unnecessary, the repeat code is not much

Comment (lerry-lee) [reply .azure-pipelines/baseline_test/baseline.test.mgmt.public.master.yml:37]: I'm not quite sure, can't we define the variables in parent stage? Like this:
https://github.com/sonic-net/sonic-buildimage/blob/eba6ef0aa9a15f72ffdc83045f5d17730a9a7ed3/azure-pipelines.yml#L86

Comment (xwjiang-ms) [reply .azure-pipelines/baseline_test/baseline.test.mgmt.public.master.yml:37]: This is defined in stage level, just like I defined

Comment (lerry-lee) [reply .azure-pipelines/baseline_test/baseline.test.mgmt.public.master.yml:37]: Oh, my bad, I regard it as job carelessly.

**Review by lerry-lee — APPROVED** [review]

**Pull request merged**